lib/vite_rails/manifest.rb

```diff
@@ -106,7 +109,7 @@ def extension_for_type(entry_type)
     case entry_type
     when :javascript then 'js'
     when :stylesheet then 'css'
-    when :typescript then dev_server_running? ? 'ts' : 'js'
+    when :typescript then 'ts'
```

Comment:
Vite now uses the actual file extension in the manifest, so we can simplify here.